libs/utils/target_script.py

```diff
@@ -0,0 +1,84 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2015, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import ast
+import inspect
+
+SCRIPT_NAME = 'remote_script.sh'
+
+class TargetScript(object):
+    """
+    This class provides utility to create and run a script
+    directly on a devlib target.
+    """
+
+    _target_attrs = ['screen_resolution', 'android_id', 'abi', 'os_version', 'model']
+
+    def __init__(self, env):
```

Comment:
Actually, looking at the example in GMaps, perhaps we can be even more generic by adding an additional

```python
def setup(script_name=SCRIPT_NAME):
    self.script_name = script_name
    self.commands = []
```
In that case, the current `setup()` method should be renamed to `push()`, which is also better describing the fact that we are building and deploying the script on target to be ready for `run()`

The combination of `setup()` and `push()` would allow to create different script, perhaps for different steps of an experiment, push and run them without creating new TargetScript objects.